### DISCUSSION TITLE
How to generate triangular mesh in cylinder?

### DISCUSSION THREAD

**Discussion by lipxisi** [idaholab/moose#23852] [Q&A General] [answered]
Hello

I have learned how to generate cylinder and mesh using ConcentricCircleMeshGenerator and MeshExtruderGenerator, which generator can generate triangular mesh in cylinder?

Thanks for your answer!

**Comment by GiudGiud** [comment, marked as answer]
Hello

Do you mean a mesh made of triangular elements in 2D, and tetrahedrals or prisms in 3D?
If so you can use this generator for a 2D circle mesh, then you can extrude it the same way as before.
https://mooseframework.inl.gov/moose/source/meshgenerators/XYDelaunayGenerator.html
Guillaume
